website/pages/docs/internals/filesystem.mdx

```diff
@@ -0,0 +1,428 @@
+---
+layout: docs
+page_title: Filesystem
+sidebar_title: Filesystem
+description: |-
+  Nomad creates an allocation working directory for every allocation. Learn what
+  goes into the working directory and how it interacts with Nomad task drivers.
+---
+
+# Filesystem
+
+Nomad creates a working directory for each allocation on a client. This
+directory can be found in the Nomad [`data_dir`] at
+`./allocs/:alloc_id`. The allocation working directory is where Nomad
+creates task directories and directories shared between tasks, write logs for
+tasks, and downloads artifacts or templates.
+
+An allocation with two tasks (named `task1` and `task2`) will have an
+allocation directory like the one below.
+
+```shell-session
+.
+├── alloc
+│   ├── data
+│   ├── logs
+│   │   ├── task1.stderr.0
+│   │   ├── task1.stdout.0
+│   │   ├── task2.stderr.0
+│   │   └── task2.stdout.0
+│   └── tmp
+├── task1
+│   ├── local
+│   ├── secrets
+│   └── tmp
+└── task2
+    ├── local
+    ├── secrets
+    └── tmp
+```
+
+-  `alloc/`: This directory is shared across all tasks in an allocation and
+  can be used to store data that needs to be used by multiple tasks, such as a
+  log shipper. This is the directory that's provided to the task as the
+  `NOMAD_ALLOC_DIR`. Note that this `alloc/` directory is not the same as the
+  "allocation working directory", which is the top-level directory. All tasks
+  in a task group can read and write to the `alloc/` directory. Within the
+  `alloc/` directory are three standard directories:
+
+  - `alloc/data/`: This directory is the location used by the
+    [`ephemeral_disk`] stanza for shared data.
+
+  - `alloc/logs/`: This directory is the location of the log files for every
+    task within an allocation. The `nomad alloc logs` command streams these
+    files to your terminal.
+
+  - `alloc/tmp/`: A temporary directory used as scratch space by task drivers.
+
+- `task/`: Each task has a **task working directory** with the same name as
+  the task. Tasks in a task group can't read each other's task working
+  directory. Depending on the task driver's [filesystem isolation mode], a
+  task may not be able to access the task working directory. Within the
+  `task/` directory are three standard directories:
+
+  - `task/local`: This directory is the location provided to the task as the
+    `NOMAD_TASK_DIR`. Note this is not the same as the "task working
+    directory". This directory is private to the task.
+
+  - `task/secrets`: This directory is the location provided to the task as
+  `NOMAD_SECRETS_DIR`. The contents of files in this directory cannot be read
+  the the `nomad alloc fs` command. It can be used to store secret data that
+  should not be visible outside the task.
+
+  - `task/tmp/`: A temporary directory used as scratch space by task drivers.
```

Comment:
why does tmp have a trailing /  (is it critical?)
```suggestion
  - `task/tmp`: A temporary directory used as scratch space by task drivers.
```

Reply:
I put the trailing `/` on all the directories to disambiguate from files, but it's not necessary in this context. Will fix for all of them.

Reply:
Fixed in 38cdafa